esmvalcore/preprocessor/_multimodel.py

```diff
@@ -248,115 +128,179 @@ def _unify_time_coordinates(cubes):
         cube.coord('time').guess_bounds()
 
 
-def _get_time_slice(cubes, time):
-    """Fill time slice array with cubes' data if time in cube, else mask."""
-    time_slice = []
-    for cube in cubes:
-        cube_time = cube.coord('time').points
-        if time in cube_time:
-            idx = int(np.argwhere(cube_time == time))
-            subset = cube.data[idx]
-        else:
-            subset = np.ma.empty(list(cube.shape[1:]))
-            subset.mask = True
-        time_slice.append(subset)
-    return time_slice
+def _time_coords_are_aligned(cubes):
+    """Return `True` if time coords are aligned."""
+    first_time_array = cubes[0].coord('time').points
 
+    for cube in cubes[1:]:
+        other_time_array = cube.coord('time').points
+        if not np.array_equal(first_time_array, other_time_array):
```

Comment:
they may differ at very low orders, since those are points in decimal format - maybe use np.allclose()?

Reply:
The CubeList won't concatenate/merge if the points differ even a little bit. I guess that is the purpose of this check?

Reply:
right! I was out of context when I made the comment above, but yes, that'll go in `_align` then into `_compute_eager` so perfect alignment is indeed needed for merging. But if you plop an `allclose()` with a decent `rtol` here and if that's True then you can switch off all the other alignment jazz and just use eg first cube's time axis as reference?

Reply:
maybe not worth the effort since those are cheap computations I reckon

Reply:
If that `allclose()` is true, you'd still need to somehow overwrite the other cubes with the reference cube, which means more conditionals, more messing with the cubes, more possibilities for errors.. 

Accomodating input data/coords with slightly (in the order of numerical precision) deviating values and treat them as equal sound to me like we should have a separate discussion about that, and not just on MM-stats.